tools/generator/Java.Interop.Tools.Generator.CodeGeneration/EnumGenerator.cs

```diff
@@ -0,0 +1,107 @@
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using static MonoDroid.Generation.EnumMappings;
+
+namespace MonoDroid.Generation
+{
+	class EnumGenerator
+	{
+		protected TextWriter sw;
+
+		public EnumGenerator (TextWriter writer)
+		{
+			sw = writer;
+		}
+
+		public void WriteEnumeration (KeyValuePair<string, EnumDescription> enu, GenBase [] gens)
+		{
+			string ns = enu.Key.Substring (0, enu.Key.LastIndexOf ('.')).Trim ();
+			string enoom = enu.Key.Substring (enu.Key.LastIndexOf ('.') + 1).Trim ();
+
+			sw.WriteLine ("namespace {0} {{", ns);
+			if (enu.Value.BitField)
+				sw.WriteLine ("  [System.Flags]");
+			sw.WriteLine ("  public enum {0} {{", enoom);
+
+			foreach (var member in enu.Value.Members) {
+				var managedMember = FindManagedMember (enu.Value, member.Key, gens);
+				sw.WriteLine ("    [global::Android.Runtime.IntDefinition (" + (managedMember != null ? "\"" + managedMember + "\"" : "null") + ", JniField = \"" + StripExtraInterfaceSpec (enu.Value.JniNames [member.Key]) + "\")]");
```

Comment:
Considering that ~everything else in `generator` emits *tabs* and not spaces, and enums appearing to be a lone holdout, should we update enum generation to *also* use tabs while we're splitting this out into a new file?

Reply:
Done